src/api-v1.js

```diff
@@ -99,7 +99,15 @@ api.declare({
       // instead of a more complicated structure
 
       if (x === 0 && (result.status === 'absent' || result.status === 'error')) {
-        let validUrl = await validateUrl(url, this.allowedPatterns, this.redirectLimit, this.ensureSSL, this.monitor);
+        let validUrl;
+        try {
+          validUrl = await validateUrl(url, this.allowedPatterns, this.redirectLimit, this.ensureSSL, this.monitor);
```

Comment:
why do this before we check the cache? and storage backend?

Obviously, the 99% use-case it's in the cache or the storage backend, which implies the URL is valid.

I wouldn't do any URL validity or redirect checks until you have to actually cache the object...